### PR TITLE
TST: organize and cleanup pandas/tests/groupby/test_aggregate.py

### LINKED ISSUE #18490: TST: organize pandas/tests/groupby/test_aggregate.py
might be nice to group these tests into separate classes within this file for organization purposes. The setup that is done (``setup_method``) is only done for a small subset of tests, the others don't need it.

### PR DESCRIPTION
**Pull request by alysivji**
closes #18490

Module currently has tests in 2 classes (`TestGroupByAggregate` and `TestGroupByAggregateCython`). The remaining tests are not in a class.

Also made the following changes:

#### test_agg_must_agg
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_agg_apply_corner
* made it more readable

#### test_agg_multiple_functions_too_many_lambdas
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_multi_function_flexible_mix
* made it more readable

### TestGroupByAggregateCython class

_all test methods that contain cython in the test name_

#### test_cython_agg_nothing_to_agg
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_cython_agg_return_dict
* replaced `self.df` with df initialized inside function

#### test_agg_dict_renaming_deprecation
* made it more readable

#### test_agg_nested_dicts
* replaced `pytest.raises` with `tm.assert_raises_regex`
* made it more readable

#### test_agg_item_by_item_raise_typeerror
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_agg_structs_series
* made it more readable

### CODE REVIEW

**Comment by jreback** [comment]
thanks @alysivji 

can you split things out to a new file (maybe just the moved tests)?

IOW each of these classes to a separate file, something like

``pandas/tests/groupby/aggregate/test_cython`` (2nd one)
``test_aggregate`` (1st class)
``test_other`` (3rd one)

just to make this even more organized.

be sure to add a ``__init__.py`` in the sub-dir

**Comment by alysivji** [comment]
* Split up the files
* cleaned up the imports
* changed all assert_frame/assert_series calls to `tm.assert_*`
* more cleaning up tests to increase readability
* rebased

@jreback lemme know if anything else needs to be done.

Thanks!

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        tm.assert_frame_equal(result, expected)
+
+    def test_agg_python_multiindex(self):
+        grouped = self.mframe.groupby(['A', 'B'])
+
+        result = grouped.agg(np.mean)
+        expected = grouped.mean()
+        tm.assert_frame_equal(result, expected)
+
+    def test_aggregate_str_func(self):
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py]: can you parametrize this (on by_weekday and by_mwkday)

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+                                          lambda x: x.weekday()])
+        _check_results(by_mwkday)
+
+    def test_aggregate_item_by_item(self):
+
+        df = self.df.copy()
+        df['E'] = ['a'] * len(self.df)
+        grouped = self.df.groupby('A')
+
+        # API change in 0.11
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py]: can remove this commented out

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import pandas as pd
+
+from pandas import bdate_range, DataFrame, Index, Series
+from pandas.core.groupby import DataError
+import pandas.util.testing as tm
+
+
+class TestGroupByAggregateCython(object):
+
+    def test_cythonized_aggers(self):
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py]: can you parameterize this

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        dr = bdate_range('1/1/2000', periods=50)
+        ts = Series(['A', 'B', 'C', 'D', 'E'] * 10, index=dr)
+
+        grouped = ts.groupby(lambda x: x.month)
+        summed = grouped.sum()
+        expected = grouped.agg(np.sum)
+        tm.assert_series_equal(summed, expected)
+
+    def test__cython_agg_general(self):
+        ops = [('mean', np.mean),
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py]: can you parameterize

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        for op, targop in ops:
+            result = df.groupby(labels)._cython_agg_general(op)
+            expected = df.groupby(labels).agg(targop)
+            try:
+                tm.assert_frame_equal(result, expected)
+            except BaseException as exc:
+                exc.args += ('operation: %s' % op, )
+                raise
+
+    def test_cython_agg_empty_buckets(self):
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]
@alysivji looking good. some requests to parameterize. could completely eliminate the need for the setup method in aggregate/test_aggregate.py if you want to make those into fixtures instead. can do this in another PR if you want.

**Comment by jreback** [comment]
cc @TomAugspurger I think we have to merge #18921 (when ready) before this (and rebase this on master)?

**Comment by TomAugspurger** [comment]
Yes it'd be nice to merge #18921 before this. Backporting can be tricky when files are split.

**Comment by jreback** [comment]
@alysivji but go ahead and make the changes as above when you can. we will be merging @TomAugspurger soon anyhow.

**Comment by alysivji** [comment]
@jreback Changes have been made. Also deleted the cython test class, not needed anymore since it's in a separate module.

I can rebase when @TomAugspurger 's changes get in.

Also started working on turning the `test_aggregate.py` setup into fixtures. Lots of moving pieces for one PR; will send a separate request once this is merged.

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        for (cat1, cat2), group in grouped:
+            expd.setdefault(cat1, {})[cat2] = op(group['C'])
+        exp = DataFrame(expd).T.stack(dropna=False)
+        exp.index.names = ['A', 'B']
+        exp.name = 'C'
+
+        result = op(grouped)['C']
+        if name in ['sum', 'prod']:
+            tm.assert_series_equal(result, exp)
+
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py]: yes u can collapse testit - just move everything up

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+])
+def test__cython_agg_general(op, targop):
+    df = DataFrame(np.random.randn(1000))
+    labels = np.random.randint(0, 50, size=1000).astype(float)
+
+    result = df.groupby(labels)._cython_agg_general(op)
+    expected = df.groupby(labels).agg(targop)
+    try:
+        tm.assert_frame_equal(result, expected)
+    except BaseException as exc:
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py]: yes u can take this try/except out
it’s helpful when not parametrized but not now

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+            result = grouped.agg(OrderedDict([['A', 'var'],
+                                              ['B', 'std'],
+                                              ['C', 'mean'],
+                                              ['D', 'sem']]))
+            expected = DataFrame(OrderedDict([['A', grouped['A'].var()],
+                                              ['B', grouped['B'].std()],
+                                              ['C', grouped['C'].mean()],
+                                              ['D', grouped['D'].sem()]]))
+            tm.assert_frame_equal(result, expected)
+
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py]: same - collapse this function

**Review by jreback — CHANGES_REQUESTED** [review]
looks good
couple of comments - will let u know when to rebase

**Comment by jreback** [comment]
@alysivji if you want to rebase we can get this in.

**Comment by alysivji** [comment]
@jreback all green!

Will start working on refactoring the setup method to fixtures this weekend.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @alysivji 

would like to remove the imports *inside* functions as well (and consolidate at the top-level) on your next PR.

**Pull request merged**